libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2080,6 +2096,21 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 			m_errorReporter.typeError(paramArgMap[i]->location(), msg);
 		}
 	}
+
+	if (_functionType->hasDeclaration())
+		if (auto const* funcDef = dynamic_cast<FunctionDefinition const*>(&_functionType->declaration()))
+		{
+			if (
+				!funcDef->isImplemented() &&
+				contains(resolveDirectBaseContracts(*m_scope), funcDef->annotation().contract) &&
```

Comment:
Why is it important whether it is a direct base or an indirect base?

Reply:
that's a good point. I had a wrong assumption here.